### PR TITLE
WT-9670: Instrument wiredtiger_open and close with event handler callbacks.

### PR DESCRIPTION
**Pull request by dgottlieb**
This is a POC demonstrating one idea for getting stats out of WT with the existing stats API while WT is opening and closing.

I've demonstrated this works with a test program (explaining why I bothered to get all of the tests to compile). But I know there's some formatting errors (and a lack of comments). No need to call those out until there's consensus for incorporating this idea.

### CODE REVIEW

File: src/conn/conn_api.c [hunk 1]
```diff
@@ -1115,6 +1115,10 @@ __conn_close(WT_CONNECTION *wt_conn, const char *config)
     conn->cache->eviction_dirty_trigger = 1.0;
     conn->cache->eviction_dirty_target = 0.1;
 
+    if (conn->default_session->event_handler) {
+        conn->default_session->event_handler->handle_conn_close(conn->default_session->event_handler);
+    }
+
```
Comment (dgottlieb) [inline src/conn/conn_api.c:1121]: WT closes all "external" sessions after this point.

File: src/conn/conn_api.c [hunk 2]
```diff
@@ -1139,6 +1143,10 @@ __conn_close(WT_CONNECTION *wt_conn, const char *config)
             WT_TRET(__wt_session_close_internal(s));
         }
 
+    if (conn->default_session->event_handler) {
```
Comment (dgottlieb) [inline src/conn/conn_api.c:1146]: The external sessions are closed before doing the rollback_to_stable call at shutdown. I wanted to capture stats for that as well, so I let the application know a connection is available again for use.

**Comment by ddanderson** [comment]
@dgottlieb, I'm going to be out next week, so just took a cursory look, and wanted to lend my support to this idea.  It seems useful for diagnosing various startup and shutdown issues.

**Comment by sueloverso** [comment]
@dgottlieb do you want to close/remove this branch since the other one has been merged?

**Comment by dgottlieb** [comment]
Yep, thanks for the nudge @sueloverso!

**Pull request closed without merging**